crates/ruff_server/src/trace.rs

```diff
@@ -0,0 +1,147 @@
+//! The tracing system for `ruff server`.
+//!
+//! Traces are controlled by the `logLevel` setting, along with the
+//! trace level set through the LSP. On VS Code, the trace level can
+//! also be set with `ruff.trace.server`. A trace level of `messages` or
+//! `verbose` will enable tracing - otherwise, no traces will be shown.
+//!
+//! `logLevel` can be used to configure the level of tracing that is shown.
+//! By default, `logLevel` is set to `"info"`.
+//!
+//! The server also supports the `RUFF_TRACE` environment variable, which will
+//! override the trace value provided by the LSP client. Use this if there's no good way
+//! to set the trace value through your editor's configuration.
+//!
+//! Tracing will write to `stderr` by default, which should appear in the logs for most LSP clients.
+//! A `logFile` path can also be specified in the settings, and output will be directed there instead.
```

Comment:
It might be useful to provide this information in the server docs either in the global docs (`crates/ruff_server/README.md`) or editor specific docs (`crates/ruff_server/docs/setup/*.md`).

Reply:
I'll update the docs in a follow-up.